Server/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp

```diff
@@ -1013,43 +1013,30 @@ int CLuaElementDefs::getElementsWithinColShape(lua_State* luaVM)
     return 1;
 }
 
-int CLuaElementDefs::getElementsWithinRange(lua_State* luaVM)
+CElementResult CLuaElementDefs::getElementsWithinRange(CVector pos, float radius, std::optional<std::string> type,
+    std::optional<unsigned short> interior, std::optional<unsigned short> dimension)
 {
-    CVector position;
-    float   radius;
-    SString elementType;
+    const auto typeHash = (type.has_value() && !type.value().empty()) ?
+        CElement::GetTypeHashFromString(type.value()) : 0;
+        
+    CElementResult result;
+    GetSpatialDatabase()->SphereQuery(result, CSphere{ pos, radius });
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadVector3D(position);
-    argStream.ReadNumber(radius);
-    argStream.ReadString(elementType, "");
-
-    if (!argStream.HasErrors())
-    {
-        // Query the spatial database
-        CElementResult result;
-        GetSpatialDatabase()->SphereQuery(result, CSphere{position, radius});
+    // Remove elements that do not match the criterias
+    result.erase(std::remove_if(result.begin(), result.end(), [&](CElement* pElement) {
```

Comment:
This should only run if `typeHash != 0 || interior.has_value() || dimension.has_value()` is true